hive/src/test/java/org/apache/iceberg/hive/HiveTableBaseTest.java

```diff
@@ -126,93 +116,91 @@ public void cleanup() {
     }
   }
 
-  private HiveConf hiveConf(Configuration conf, int port) {
-    final HiveConf hiveConf = new HiveConf(conf, this.getClass());
-    hiveConf.set(THRIFT_URIS.getVarname(), "thrift://localhost:" + port);
-    hiveConf.set(WAREHOUSE.getVarname(), "file:" + hiveLocalDir.getAbsolutePath());
-    hiveConf.set(WAREHOUSE.getHiveName(), "file:" + hiveLocalDir.getAbsolutePath());
-    hiveConf.set(CONNECT_URL_KEY.getVarname(), "jdbc:derby:" + getDerbyPath() + ";create=true");
+  HiveTables tables;
+
+  @Before
+  public void createTestTable() throws Exception {
+    this.tables = new HiveTables(hiveConf);
+    tables.create(schema, partitionSpec, DB_NAME, TABLE_NAME);
+  }
+
+  @After
+  public void dropTestTable() throws Exception {
+    metastoreClient.dropTable(DB_NAME, TABLE_NAME);
+    tables.close();
+    this.tables = null;
+  }
+
+  private static HiveConf hiveConf(Configuration conf, int port) {
+    final HiveConf hiveConf = new HiveConf(conf, HiveTableBaseTest.class);
+    hiveConf.set(HiveConf.ConfVars.METASTOREURIS.varname, "thrift://localhost:" + port);
+    hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, "file:" + hiveLocalDir.getAbsolutePath());
     return hiveConf;
   }
 
-  private String getDerbyPath() {
+  private static String getDerbyPath() {
     final File metastore_db = new File(hiveLocalDir, "metastore_db");
     return metastore_db.getPath();
   }
 
-  private TServer thriftServer() throws IOException,
-          TTransportException,
-          MetaException,
-          InvocationTargetException,
-          NoSuchMethodException,
-          IllegalAccessException,
-          NoSuchFieldException {
-    final TServerSocketKeepAlive socket = new TServerSocketKeepAlive(new TServerSocket(0));
-    this.hiveConf = hiveConf(new Configuration(), socket.getServerSocket().getLocalPort());
-    HiveMetaStore.HMSHandler baseHandler = new HiveMetaStore.HMSHandler("new db based metaserver", hiveConf);
-    IHMSHandler handler = RetryingHMSHandler.getProxy(hiveConf, baseHandler, true);
-    final TTransportFactory transportFactory = new TTransportFactory();
-    final TSetIpAddressProcessor<IHMSHandler> processor = new TSetIpAddressProcessor<>(handler);
+  private static TServer thriftServer() throws Exception {
+    TServerSocket socket = new TServerSocket(0);
+    HiveTableBaseTest.hiveConf = hiveConf(new Configuration(), socket.getServerSocket().getLocalPort());
+    HiveConf serverConf = new HiveConf(hiveConf);
+    serverConf.set(HiveConf.ConfVars.METASTORECONNECTURLKEY.varname, "jdbc:derby:" + getDerbyPath() + ";create=true");
+    HiveMetaStore.HMSHandler baseHandler = new HiveMetaStore.HMSHandler("new db based metaserver", serverConf);
+    //IHMSHandler handler = RetryingHMSHandler.getProxy(serverConf, baseHandler, false);
```

Comment:
remove?

Reply:
This was an unnecessary change I introduced while trying to find out why the tests were failing. I'll revert it.